src/kernel/pvclock.c

```diff
@@ -5,6 +5,19 @@
 BSS_RO_AFTER_INIT static heap pvclock_heap;
 BSS_RO_AFTER_INIT static volatile struct pvclock_vcpu_time_info *vclock;
 
+static timestamp pvclock_get_rtc_offset(struct pvclock_wall_clock *wc)
```

Comment:
Maybe it doesn't matter because of the read_barriers acting as compiler barriers, but it seems that the struct at wc should be treated as volatile.

Reply:
OK, changed the parameter of this function to `volatile struct pvclock_wall_clock *wc`